api_examples/hsfs/feature_group_change_notification_cdc.ipynb

```diff
@@ -0,0 +1,220 @@
+      "\n",
+      "Logged in to project, explore it here https://snurran.hops.works/p/5243\n",
+      "Connected. Call `.close()` to terminate connection gracefully.\n"
+     ]
+    }
+   ],
+   "source": [
+    "import hopsworks\n",
+    "import pandas as pd\n",
+    "\n",
+    "project = hopsworks.login()\n",
+    "fs = project.get_feature_store()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4103a7d4",
+   "metadata": {},
+   "source": [
+    "### Create the CDC Kafka Topic for Notifications"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "77225f95",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Topic already exists with name: jim_notification\n"
+     ]
+    }
+   ],
+   "source": [
+    "feature_group_name = \"cdc\"\n",
+    "notification_topic =f\"{feature_group_name}_notification\"\n",
+    "\n",
+    "if not any(topic.name == notification_topic for topic in project.get_kafka_api().get_topics()):\n",
+    "    project.get_kafka_api().create_topic(notification_topic, None, None)\n",
+    "else:\n",
+    "    print(f\"Topic already exists with name: {notification_topic}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ce88012",
+   "metadata": {},
+   "source": [
+    "### Create the Feature Group, providing the CDC topic name "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d4899e28",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fg = fs.get_or_create_feature_group(name=feature_group_name, \n",
+    "                                    version=1, \n",
+    "                                    primary_key=[\"id\"], \n",
+    "                                    online_enabled=True, \n",
+    "                                    notification_topic_name=notification_topic)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "964e166d",
+   "metadata": {},
+   "source": [
+    "### Insert Data into the Feature Group"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fd384d3b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "size = 10\n",
+    "fg_data = {'id': range(0, size), 'text': \"test\"}\n",
+    "fg_df = pd.DataFrame.from_dict(fg_data)\n",
+    "\n",
+    "fg.insert(fg_df)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4e156aa",
+   "metadata": {},
+   "source": [
+    "### Create a Kafka Consumer to receive the Feature Group notifications"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "89adcc86",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from confluent_kafka import Consumer\n",
+    "\n",
+    "storage_connector = fs._storage_connector_api.get_kafka_connector(fs.id)\n",
+    "\n",
+    "config = storage_connector.confluent_options()\n",
+    "config[\"auto.offset.reset\"] = \"earliest\"\n",
+    "config[\"group.id\"] = feature_group_name\n",
```

Comment:
the issue with empty poll is fixed in the notebooks that i sent you.

Reply:
Fixed.